### ISSUE TITLE
entry_to_ldif() tries to enumerate None.

### ISSUE THREAD

**Issue by Jo-Con-El** [cannatag/ldap3#689]
Version: ldap3 2.6
OS: MacOS 10.14

I retrieved all the AD users in an OU with a Reader object as usual:
```
>>> obj_user = ldap3.ObjectDef('user', conn)
>>> r = ldap3.Reader(conn, obj_user, "ou=Users, dc=company, dc=internal")
>>> r.search()
[...]
>>> r
CURSOR : Reader
CONN   : ldaps://10.20.30.40:636 - ssl - user: company\user - not lazy - bound - open - <local: 10.10.10.10:58588 - remote: 10.20.30.40:636> - tls not started - listening - SyncStrategy - internal decoder
DEFS   : user
ATTRS  : ['USNIntersite', 'aCSPolicyName', 'accountExpires', 'adminCount', 'adminDescription', 'adminDisplayName', 'allowedAttributes', 'allowedAttributesEffective', [...], 'x500uniqueIdentifier']
BASE   : 'ou=Users, dc=company, dc=internal' [SUB]
FILTER : '(objectClass=user)'
ENTRIES: 477 [executed at: 2019-05-09T14:01:58.136922]
```

The entries are apparently valid:
```
>>> r[0]
DN: CN=Albino Aardvark,OU=Users,DC=company,DC=internal - STATUS: Read - READ TIME: 2019-05-09T14:01:49.374891
    accountExpires: 9999-12-31 23:59:59.999999
    allowedAttributes: shadowFlag
                       shadowExpire
                       shadowInactive
                       shadowWarning
[...]
    userAccountControl: 512
    userPrincipalName: aaardvark@company.internal
    whenChanged: 2019-01-28 23:17:03+00:00
    whenCreated: 2017-06-09 02:41:46+00:00
```

But using `entry_to_ldif()` crashes with a _NoneType not iterable_ error:
```
>>> r[0].entry_to_ldif()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/ldap3/abstract/entry.py", line 339, in entry_to_ldif
    ldif_lines = operation_to_ldif('searchResponse', [self._state.response], all_base64, sort_order=sort_order)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/ldap3/protocol/rfc2849.py", line 209, in operation_to_ldif
    lines = search_response_to_ldif(entries, all_base64, sort_order)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/ldap3/protocol/rfc2849.py", line 129, in search_response_to_ldif
    lines.extend(add_attributes(entry['raw_attributes'], all_base64))
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/ldap3/protocol/rfc2849.py", line 112, in add_attributes
    for val in attributes[attr]:
TypeError: 'NoneType' object is not iterable
```

`r[0].entry_to_json()` works fine, though.

**Comment by cfelder** [comment]
Hi there. After some investigation on a bug report in ldap3-orm I found a similar issue using a Writer cursor for creating a new entry, see minimal crashing example below.

```python
from ldap3 import Connection, ObjectDef, Writer
from ldap3.utils.dn import safe_dn

base_dn = "dc=example,dc=com"
conn = Connection(...)


w = Writer(conn, ObjectDef('inetOrgPerson', conn))
w.uid = 'john'
w.givenName = 'John'
w.sn = 'Doe'
w.cn = 'John Doe'
entry = w.new(safe_dn(('uid={uid},ou=People,' + base_dn).format(uid=w.uid)))
entry.entry_to_ldif()  # fails
```

Stacktrace:
```python
TypeError                                 Traceback (most recent call last)
<ipython-input-19-269fbb85b67b> in <module>
----> 1 entry.entry_to_ldif()

/usr/local/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/ldap3/abstract/entry.py in entry_to_ldif(self, all_base64, line_separator, sort_order, stream)
    337 
    338     def entry_to_ldif(self, all_base64=False, line_separator=None, sort_order=None, stream=None):
--> 339         ldif_lines = operation_to_ldif('searchResponse', [self._state.response], all_base64, sort_order=sort_order)
    340         ldif_lines = add_ldif_header(ldif_lines)
    341         line_separator = line_separator or linesep

/usr/local/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/ldap3/protocol/rfc2849.py in operation_to_ldif(operation_type, entries, all_base64, sort_order)
    207 def operation_to_ldif(operation_type, entries, all_base64=False, sort_order=None):
    208     if operation_type == 'searchResponse':
--> 209         lines = search_response_to_ldif(entries, all_base64, sort_order)
    210     elif operation_type == 'addRequest':
    211         lines = add_request_to_ldif(entries, all_base64, sort_order)

/usr/local/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/ldap3/protocol/rfc2849.py in search_response_to_ldif(entries, all_base64, sort_order)
    125     lines = []
    126     for entry in entries:
--> 127         if 'dn' in entry:
    128             lines.append(_convert_to_ldif('dn', entry['dn'], all_base64))
    129             lines.extend(add_attributes(entry['raw_attributes'], all_base64))

TypeError: argument of type 'NoneType' is not iterable
```

ldap3 version is 2.6 as well.

**Comment by cannatag** [comment]
Hi, can you try the code in dev? It should work now. Thanks.

**Comment by cfelder** [comment]
Hi Giovanni. I can confirm that my example doesn't crash anymore and ``entry_to_ldif`` returns an empty string ``''`` now. It resolves this issue but would it be possible to return the ldif representation for new entries?

It is possible using a fake connection instance ``conn = Connection(server=None, client_strategy=LDIF)`` and a ``conn.add`` operation. It must be possible on client side though.

See my first comment in issue cfelder/ldap3-orm#2

Best
Christian

**Comment by cannatag** [comment]
released in v2.6.1
